.github/workflows/ci.yml

```diff
@@ -17,44 +17,44 @@ concurrency:
 jobs:
   build:
     name: Build
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
 
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
     - name: Generate metadata
       id: meta
-      uses: docker/metadata-action@v3
+      uses: docker/metadata-action@v5
       with:
         images: |
           docker.io/zephyrprojectrtos/scancode
           ghcr.io/zephyrproject-rtos/scancode
 
     - name: Set up QEMU
-      uses: docker/setup-qemu-action@v1
+      uses: docker/setup-qemu-action@v3.6.0
```

Comment:
Any reason to specify the action version down to the rev number?

Unless we are pinning for security reasons, in which case we should be using the commit sha, we should just use `@v3`.